compiler/rustc_data_structures/src/marker.rs

```diff
@@ -201,24 +201,31 @@ pub fn assert_dyn_send_val<T: ?Sized + DynSend>(_t: &T) {}
 pub fn assert_dyn_send_sync_val<T: ?Sized + DynSync + DynSend>(_t: &T) {}
 
 #[derive(Copy, Clone)]
-pub struct IsSend<T: DynSend>(pub T);
+pub struct FromDyn<T>(T);
+
+impl<T> FromDyn<T> {
+    #[inline(always)]
+    pub fn from(val: T) -> Self {
+        #[cfg(parallel_compiler)]
+        assert!(crate::sync::active());
+        FromDyn(val)
+    }
 
-impl<T: DynSend> IsSend<T> {
     #[inline(always)]
     pub fn into_inner(self) -> T {
         self.0
     }
 }
 
-// `IsSend` is `Send` since it check when created.
+// `FromDyn` is `Send` if `T` is `Send`, since it check when created.
 #[cfg(parallel_compiler)]
-unsafe impl<T: DynSend> std::marker::Send for IsSend<T> {}
+unsafe impl<T: DynSend> std::marker::Send for FromDyn<T> {}
```

Comment:
You can just import `Send` and `Sync` now, as there's no conflict with the previous dummy trait names.

Reply:
Done :)